### PR TITLE
Refactoring & dynamic pagination handling for site fetching & ..

### PR DESCRIPTION
**Pull request by EvaX3me**

### CODE REVIEW

File: geo_map/static/geo_map/helpers.js [hunk 1]
```diff
@@ -173,16 +173,14 @@ function debounce(func, wait) {
...
-      validSites.push(site);
-    }
+  const validSites = sites.filter(site => site.lat && site.lng);
 
-    if (validSites.length === 2) {
-      break;
+  validSites.sort((a, b) => {
+    if (a.lat === b.lat) {
+      return a.lng - b.lng;
     }
```
Comment (ManiacMaxo) [inline geo_map/static/geo_map/helpers.js:181]: why is this here?

File: geo_map/static/geo_map/index.js [hunk 2]
```diff
@@ -60,14 +61,88 @@ document.addEventListener("DOMContentLoaded", function () {
...
+  const currentCenter = map ? map.getCenter() : null;
+  storedMapCenter = currentCenter;
+  storedZoomLevel = currentZoom;
+
+  const SITES_API_CALL = new URL(baseURL + "/api/plugins/geo_map/sites/");
+  const LINKS_API_CALL = new URL(baseURL + "/api/plugins/geo_map/links/");
+
+  if (selectedPopsStatuses.length) {
+    SITES_API_CALL.searchParams.set("status__in", selectedPopsStatuses.join(","));
+    SITES_API_CALL.search = SITES_API_CALL.searchParams.toString();
```
Comment (ManiacMaxo) [inline geo_map/static/geo_map/index.js:89]: No need for setting search here

File: geo_map/static/geo_map/index.js [hunk 3]
```diff
@@ -60,14 +61,88 @@ document.addEventListener("DOMContentLoaded", function () {
...
+    const [sitesResponse, linksResponse] = await Promise.all([
+      fetch(SITES_API_CALL),
+      fetch(LINKS_API_CALL)
+    ]);
+
+    const sitesData = await sitesResponse.json();
+    const linksData = await linksResponse.json();
+
+    allSites = JSON.parse(JSON.stringify(sitesData));
+    allLinks = JSON.parse(JSON.stringify(linksData));
```
Comment (ManiacMaxo) [inline geo_map/static/geo_map/index.js:112]: Why do you need to copy the data?

**Review by ManiacMaxo — COMMENTED** [review]
Some questions

File: geo_map/static/geo_map/index.js (hunk 2 above)
Comment (EvaX3me) [reply geo_map/static/geo_map/index.js:89]: where should be added ?

File: geo_map/static/geo_map/index.js [hunk 4]
```diff
@@ -60,14 +61,86 @@ document.addEventListener("DOMContentLoaded", function () {
...
+  storedZoomLevel = currentZoom;
+
+  const SITES_API_CALL = new URL(baseURL + "/api/plugins/geo_map/sites/");
+  const LINKS_API_CALL = new URL(baseURL + "/api/plugins/geo_map/links/");
+
+  if (selectedPopsStatuses.length) {
+    SITES_API_CALL.searchParams.set("status__in", selectedPopsStatuses.join(","));
+  }
+  if (selectedGroups.length && selectedPopsStatuses.length) {
+    SITES_API_CALL.searchParams.set("?group__in", selectedGroups.join(","));
```
Comment (ManiacMaxo) [inline geo_map/static/geo_map/index.js:91]: typo? remove `?`

File: geo_map/static/geo_map/index.js [hunk 5]
```diff
@@ -60,14 +61,86 @@ document.addEventListener("DOMContentLoaded", function () {
...
+  }
+  if (selectedTenants.length) {
+    LINKS_API_CALL.searchParams.set("provider__in", selectedTenants.join(","));
+  }
+  if (selectedFiberLinkStatuses.length) {
+    LINKS_API_CALL.searchParams.set("?status__in", selectedFiberLinkStatuses.join(","));
```
Comment (ManiacMaxo) [inline geo_map/static/geo_map/index.js:97]: `?`

File: geo_map/static/geo_map/index.js (hunk 3 above)
Comment (ManiacMaxo) [reply geo_map/static/geo_map/index.js:110]: I meant the `JSON.parse(JSON.stingify(...))`

File: geo_map/static/geo_map/index.js (hunk 4 above)
Comment (EvaX3me) [reply geo_map/static/geo_map/index.js:91]: I can't remove group__in because the API requires it to properly filter by groups, and removing it causes an error due to invalid query parameters
